Assets/MixedRealityToolkit.Providers/ObjectMeshObserver/RoomFileFormat/RoomFileImporter.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using UnityEditor.Experimental.AssetImporters;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SpatialObjectMeshObserver.RoomFile
+{
+    [ScriptedImporter(1, "room")]
+    public class RoomFileImporter : ScriptedImporter
+    {
+        public override void OnImportAsset(AssetImportContext context)
+        {
+            FileInfo fileInfo = new FileInfo(context.assetPath);
+            string name = fileInfo.Name.Split(new char[] { '.' })[0];
+
+            IList<Mesh> meshes;
+
+            using (BinaryReader reader = OpenFileForRead(fileInfo.FullName))
+            {
+                meshes = RoomFileSerializer.Deserialize(reader);
+            }
+
+            GameObject model = new GameObject(name);
+            context.AddObjectToAsset(name, model);
+
+            int i = 0;
+            foreach (Mesh mesh in meshes)
```

Comment:
Since we're keeping track of an `i` here anyway (and incrementing it each loop), does it make sense (is it possible) to just use a `for` loop here?

Reply:
good call!